classes/MembershipExpirationProcessor.js

```diff
@@ -0,0 +1,59 @@
+const { Op } = require('sequelize')
+
+module.exports = class MembershipExpirationProcessor {
+  constructor ({ api, config }) {
+    this.api = api
+    this.emailTemplateName = config.emailTemplateName
+    this.emailSubject = config.emailSubject
+  }
+
+  async processMemberships (fromDate, toDate) {
+    if (!(fromDate && toDate)) {
+      return new Error('Provide time period!')
+    }
+
+    const exipiringMemberships = await this.api.models.member.findAll({
+      where: { membershipEndDate: { [Op.between]: [fromDate, toDate] } }
+    })
+
+    const results = exipiringMemberships.map(async (member) => {
+      if (member.notifiedForExpiringDate === member.membershipEndDate) {
```

Comment:
you can add this check to the query using this [SO](https://stackoverflow.com/a/35665536/533873)

Reply:
Already tried that, maybe because of missing knowledge couldn't make it to work..